tools/file_packager.py

```diff
@@ -738,12 +777,51 @@ def was_seen(name):
   });
 '''
 
-ret += '''
+ret += '''%s
 })();
-'''
+''' % ('''
+  Module['removeRunDependency']('%(metadata_file)s');
+ }
+
+ var REMOTE_METADATA_NAME = typeof Module['locateFile'] === 'function' ?
+                            Module['locateFile']('%(metadata_file)s') :
+                            ((Module['filePackagePrefixURL'] || '') + '%(metadata_file)s');
+ var xhr = new XMLHttpRequest();
+ xhr.onreadystatechange = function() {
+  if (xhr.readyState == 4 && xhr.status == 200) {
+    loadPackage(JSON.parse(xhr.responseText));
+  }
+ }
+ xhr.open('GET', REMOTE_METADATA_NAME, true);
+ xhr.overrideMimeType('application/json');
+ xhr.send(null);
+
+ if (!Module['preRun']) Module['preRun'] = [];
+ Module["preRun"].push(function() {
+  Module['addRunDependency']('%(metadata_file)s');
```

Comment:
These **new** codes are emitted to the js output file. I have a dilemma here. Currently, top level has no indentation and level 1 has a 2-space indentation. When using ```separate-metadata``` option, the new codes are emitted in between the existing top level and level 1. If I makes my codes to become the level 1 then I have to shift the existing level 1 to level 2 (and that all over the places!). And the end result does not look pretty either for user who does not use the ```separate-metadata``` option later.